jujugui/static/gui/src/app/jujulib/charmstore.js

```diff
@@ -504,6 +504,51 @@ var module = module;
         });
         callback(null, truncatedList);
       }
+    },
+
+    /**
+      Gets the list of resources for the supplied charm.
+
+      @method getResources
+      @param charmId {String} The id of the charm for which to get resources.
+      @params callback {Function} A callback to handle errors or accept the data
+          from the request. Must accept an error message or null as its first
+          parameter and the response data as its second.
+    */
+    getResources: function(charmId, callback) {
+      charmId = charmId.replace('cs:', '');
+      return jujulib._makeRequest(
+          this.bakery,
+          this._generatePath(charmId, null, '/meta/resources'),
+          'GET',
+          null,
+          this._processResources.bind(this, callback));
+    },
+
+    /**
+      Process the data returned by the call to get charm resources.
+
+      @method _processResources
+      @param callback {Function} A callback to handle errors or accept the data
+          from the request. Must accept an error message or null as its first
+          parameter and the response data as its second.
+      @param error {String} An error message or null if no error.
+      @param data {Object} The response from the request.
+    */
+    _processResources: function(callback, error, data) {
+      if (error !== null) {
+        callback(error, null);
```

Comment:
I'm partial to including the data we have when there is an error. Then the client of the lib can make it's own choice what to do with all the information it has.